src/file/metadata.rs

```diff
@@ -354,8 +374,391 @@ impl ColumnChunkMetaData {
       total_uncompressed_size,
       data_page_offset,
       index_page_offset,
-      dictionary_page_offset
+      dictionary_page_offset,
+      statistics
     };
     Ok(result)
   }
 }
+
+/// Column order to compare values with statistics min/max.
+#[derive(Clone, Debug, PartialEq)]
+pub enum ColumnOrder {
+  /// Signed (either value or legacy byte-wise) comparison.
+  SIGNED,
+  /// Unsigned (depending on physical type either value or byte-wise) comparison.
+  UNSIGNED,
+  /// Comparison is undefined.
+  UNDEFINED
+}
+
+/// Optional column statistics that are used per row group and per page.
+///
+/// Use `order` field to determine comparison order for a type.
+/// When statistics contain deprecated min/max fields, then comparison is always signed.
+#[derive(Clone, Debug, PartialEq)]
+pub enum Statistics {
+  Boolean { order: ColumnOrder, min: Option<bool>, max: Option<bool>, nulls: u64 },
+  Int32 { order: ColumnOrder, min: Option<i32>, max: Option<i32>, nulls: u64 },
+  Int64 { order: ColumnOrder, min: Option<i64>, max: Option<i64>, nulls: u64 },
+  Int96 { order: ColumnOrder, min: Option<Int96>, max: Option<Int96>, nulls: u64 },
+  Float { order: ColumnOrder, min: Option<f32>, max: Option<f32>, nulls: u64 },
+  Double { order: ColumnOrder, min: Option<f64>, max: Option<f64>, nulls: u64 },
+  Binary { order: ColumnOrder, min: Option<Vec<u8>>, max: Option<Vec<u8>>, nulls: u64 }
+}
+
+impl Statistics {
+  /// Returns number of nulls for the column.
+  /// Note that even though statistics are for leaf columns, null count also takes into
+  // account null complex types, e.g. lists.
+  pub fn null_count(&self) -> u64 {
+    match *self {
+      Statistics::Boolean { nulls, ..} => nulls,
+      Statistics::Int32 { nulls, ..} => nulls,
+      Statistics::Int64 { nulls, ..} => nulls,
+      Statistics::Int96 { nulls, ..} => nulls,
+      Statistics::Float { nulls, ..} => nulls,
+      Statistics::Double { nulls, ..} => nulls,
+      Statistics::Binary { nulls, ..} => nulls
+    }
+  }
+
+  /// Returns `true` if column contains at least one null, `false` otherwise.
+  pub fn has_nulls(&self) -> bool {
+    self.null_count() > 0
+  }
+
+  /// Method to convert from Thrift definition.
+  /// Note that column type should match statistics, otherwise, there is a risk of
+  /// invalid conversion.
+  pub fn from_thrift(
```

Comment:
do we also need to process `column_orders` from `FileMetaData`?

Reply:
Yes, I will update.

Reply:
Based on my comment above, `FileMetaData` will have a method that returns column orders for leaf columns, and sort orders could be extracted from it.